agent/dehydrate.js

```diff
@@ -11,6 +11,60 @@
 'use strict';
 
 /**
+ * Get a enhanced/articial type string based on the object instance
+ */
+function getPropType(data: Object): string | null {
+  if (!data) {
+    return null;
+  }
+  if (data._reactFragment) {
+    return 'react_fragment';
+  }
+  if (Array.isArray(data)) {
+    return 'array';
+  }
+  if (ArrayBuffer.isView(data)) {
+    if (data instanceof DataView) {
+      return 'data_view';
+    }
+    return 'typed_array';
+  }
+  if (data instanceof ArrayBuffer) {
+    return 'array_buffer';
+  }
+
+  var type = typeof data;
```

Comment:
I think I have a slight preference for doing this first, as all of the above checks are only if `typeof -> 'object'`